File: src/core/priority_fee/fixed_fee.py
from . import PriorityFeePlugin


class FixedPriorityFee(PriorityFeePlugin):
    """Fixed priority fee plugin."""

    def __init__(self, fixed_fee: int):
        """
        Initialize the fixed fee plugin.

        Args:
            fixed_fee: Fixed priority fee in microlamports.
        """
        self.fixed_fee = fixed_fee

    async def get_priority_fee(self) -> int | None:
        """
        Return the fixed priority fee.

        Returns:
            Optional[int]: Fixed priority fee in microlamports, or None if fixed_fee is 0.
        """
        if self.fixed_fee == 0:
            return None
        return self.fixed_fee

File: src/core/priority_fee/fixedFee.ts
import { PriorityFeePlugin } from './manager'; // Assuming PriorityFeePlugin is in manager.ts or a base file

export class FixedPriorityFee implements PriorityFeePlugin {
  private fixedFee: number;

  /**
   * Initialize the fixed fee plugin.
   * @param fixedFee Fixed priority fee in microlamports.
   */
  constructor(fixedFee: number) {
    this.fixedFee = fixedFee;
  }

  /**
   * Return the fixed priority fee.
   * @returns Fixed priority fee in microlamports, or null if fixedFee is 0.
   */
  public async getPriorityFee(): Promise<number | null> {
    if (this.fixedFee === 0) {
      return null;
    }
    return this.fixedFee;
  }
}
